synapse/handlers/federation_event.py

```diff
@@ -1220,6 +1245,26 @@ async def _handle_marker_event(self, origin: str, marker_event: EventBase) -> No
 
         logger.debug("_handle_marker_event: received %s", marker_event)
 
+        # TODO: Move this to a background queue
+        async def handle_marker_queue(marker_event: EventBase) -> None:
```

Comment:
Which queue pattern should I use? There are a few places we do `add_to_queue` stuff. I think want to do something similar to the [`_EventPeristenceQueue`](https://github.com/matrix-org/synapse/blob/409573f6d0f146db2a55914cb7b65a0a95f6fde5/synapse/storage/persist_events.py#L128) where it's able to handle a queue per-room but it looks like a copy-paste boilerplate. Can we abstract it?

Reply:
This is no longer relevant since we don't have to keep navigating up through each era of state to find the markers. Alternative: https://github.com/matrix-org/synapse/pull/12718#discussion_r876581528